packages/firestore/src/util/obj_map.ts

```diff
@@ -110,4 +110,8 @@ export class ObjectMap<KeyType, ValueType> {
   isEmpty(): boolean {
     return isEmpty(this.inner);
   }
+
+  size(): number {
+    return objectSize(this.inner);
```

Comment:
This is pretty expensive and will turn into an O(n) scan for every Overlay that we return. Can we track the size manually?